src/Diagnostics.DataProviders/DataProviders/Base/SupportObserverDataProviderBase.cs

```diff
@@ -101,9 +101,18 @@ protected async Task<string> GetObserverResource(string url, string resourceId =
                 request.Headers.TryAddWithoutValidation("Authorization", await GetToken(resourceId));
             }
 
-            var cancelToken = new CancellationToken();
-            var response = await _httpClient.SendAsync(request, cancelToken);
-            response.EnsureSuccessStatusCode();
+            var response = await _httpClient.SendAsync(request);
+
+            try
+            {
+                response.EnsureSuccessStatusCode();
+            }
+            catch (HttpRequestException ex)
+            {
+                ex.Data.Add("StatusCode", response.StatusCode);
```

Comment:
you should probably add response content string also if not already present in the exception.

Reply:
I am not sure whether is for logging purposes also. I see below you use this status code.

Reply:
I'm just using this statuscode for the api response in the controller. Response content sounds like a good idea.